### PR TITLE
[FIXED JENKINS-38993] Add deterministic ordering of conditions.

### PR DESCRIPTION
**Pull request by abayer**
[JENKINS-38993](https://issues.jenkins-ci.org/browse/JENKINS-38993)
Ordering is:
* Always
* Changed
* Aborted/Failure/Success/Unstable

This is all done via ordinals.

cc @reviewbybees esp @stephenc @rsandell

### CODE REVIEW

**Comment by stephenc** [comment]
:bee:

File: pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/model/BuildCondition.java [hunk 1]
```diff
@@ -43,13 +47,40 @@
     @Whitelisted
     public abstract boolean meetsCondition(WorkflowRun r);
 
+    public abstract double getOrdinal();
```
Comment (rsandell) [inline pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/model/BuildCondition.java]: Why not use the ordinal that `@Extension` already provides?

File: pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/model/BuildCondition.java [hunk 2]
```diff
@@ -43,13 +47,40 @@
...
+
+        List<String> toSort = new ArrayList<>(conditionMethods.keySet());
+
+        Collections.sort(toSort, new Comparator<String>() {
+            @Override
+            public int compare(String c1, String c2) {
+                BuildCondition firstCondition = conditionMethods.get(c1);
+                BuildCondition secondCondition = conditionMethods.get(c2);
+
+                if (secondCondition == null) {
```
Comment (rsandell) [inline pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/model/BuildCondition.java]: :bug: if both are `null` you should return `0`.

File: pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/model/BuildCondition.java (hunk 1 above)
Comment (abayer) [inline pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/model/BuildCondition.java]: ...good question. I straight-up didn't think about that.

File: pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/model/BuildCondition.java (hunk 2 above)
Comment (abayer) [inline pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/model/BuildCondition.java]: Good point.

**Comment by abayer** [comment]
@rsandell Thoughts now?

**Comment by rsandell** [comment]
:bee:

**Pull request merged**